ClearKeep/Scenes/Home/Models/CallViewModel.swift

```diff
@@ -0,0 +1,454 @@
+	}
+	
+	func getRemoteVideoRenderSize(videoView: RTCEAGLVideoView) -> CGSize {
+		if let renderSize = callBox?.videoRoom?.remotes.first(where: { $0.value.videoRenderView == videoView })?.value.renderSize {
+			return renderSize
+		}
+		return CGSize.zero
+	}
+	
+	private func startCallTimout() {
+		// Check timeout for call
+		timeoutTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(CallViewModel.checkCallTimeout), userInfo: nil, repeats: true)
+		guard let timeoutTimer = timeoutTimer else {
+			return
+		}
+		RunLoop.current.add(timeoutTimer, forMode: .default)
+	}
+	
+	private func startCallTimer() {
+//        if callControl.signalingState != .answered || callControl.mediaState != .connected {
+//            return
+//        }
+		
+		if callTimer == nil {
+			timeCounter.sec = 0
+			timeCounter.hour = 0
+			timeCounter.min = 0
+			// Bắt đầu đếm giây
+			callTimer = Timer(timeInterval: 1, target: self, selector: #selector(CallViewModel.timeTick(timer:)), userInfo: nil, repeats: true)
+			guard let callTimer = callTimer else {
+				return
+			}
+			RunLoop.current.add(callTimer, forMode: .default)
+			callTimer.fire()
+			
+			// => Ko check timeout nữa
+			self.stopTimeoutTimer()
+		}
+	}
+	
+	@objc private func timeTick(timer: Timer) {
+		let timeNow = timeCounter.timeNow()
+		timeCall = timeNow
+	}
+	
+	private func stopCallTimer() {
+		CFRunLoopStop(CFRunLoopGetCurrent())
+		callTimer?.invalidate()
+		callTimer = nil
+	}
+	
+	private func stopTimeoutTimer() {
+		CFRunLoopStop(CFRunLoopGetCurrent())
+		timeoutTimer?.invalidate()
+		timeoutTimer = nil
+	}
+	
+	@objc private func checkCallTimeout() {
+		print("checkCallTimeout \(callInterval)")
+		callInterval += 1
+		if callInterval >= 60 && callTimer == nil {
+			endCall()
+		}
+	}
+	
+	func updateCallTypeVideo() {
+		guard let callBox = self.callBox else { return }
+		DispatchQueue.main.async { [weak self] in
+			guard let self = self else { return }
+			self.cameraOn = true
+			self.callType = .video
+			callBox.isCamera = true
+			callBox.type = .video
+			callBox.videoRoom?.publisher?.cameraOn()
+			self.speakerEnable = true
+			self.updateSpeakerConfig()
+			print("#TEST updateCallTypeVideo >>> video type")
+		}
+	}
+	
+	func updateCallType(data: PublicationNotification) {
+		guard let callBox = self.callBox,
+			  callBox.roomRtcId == Int64(data.groupID ?? "0")
+		else {
+			return
+		}
+		
+		DispatchQueue.main.async { [weak self] in
+			guard let self = self else { return }
+			self.cameraOn = false
+			self.callType = data.notifyType == "video" ? .video : .audio
+			self.callBox?.type = self.callType == .video ? .video : .audio
+			callBox.isCamera = false
+			if self.callType == .video {
+				self.callBox?.videoRoom?.publisher?.cameraOff()
+			}
+			self.speakerEnable = true
+			self.updateSpeakerConfig()
+			print("#TEST updateCallTypeVideo >>> video type")
+		}
+	}
+	
+	func didReceiveMessageGroup(userInfo: [AnyHashable : Any]?) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/fe9f474e9dacdb67166a3b18715e917932515a0f/ClearKeep/Scenes/Home/Models/CallViewModel.swift#L377"></span>Colons should be next to the identifier when specifying a type and next to the key in dictionary literals.<br />
<code>colon</code> <code>CallViewModel.swift:377</code></td>
    </tr>
  </tbody>
</table>